vcd/resource_vcd_vapp.go

```diff
@@ -454,35 +432,35 @@ func resourceVcdVAppRead(d *schema.ResourceData, meta interface{}) error {
 }
 
 func getVAppIPAddress(d *schema.ResourceData, meta interface{}, vdc govcd.Vdc, org govcd.Org) (string, error) {
-	vcdClient := meta.(*VCDClient)
 	var ip string
-	err := retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-		vapp, err := vdc.FindVAppByName(d.Id())
-		if err != nil {
-			return resource.RetryableError(fmt.Errorf("unable to find vapp"))
-		}
 
-		// getting the IP of the specific Vm, rather than index zero.
-		// Required as once we add more VM's, index zero doesn't guarantee the
-		// 'first' one, and tests will fail sometimes (annoying huh?)
-		vm, err := vdc.FindVMByName(vapp, d.Get("name").(string))
-		if err != nil {
-			return resource.RetryableError(fmt.Errorf("unable to find vm: %s", err))
-		}
+	vapp, err := vdc.FindVAppByName(d.Id())
+	if err != nil {
+		fmt.Errorf("unable to find vapp")
```

Comment:
There should be a `return` statement here

Reply:
fixed